### PR TITLE
Class for Optimum-Intel Embedding Model

### PR DESCRIPTION
**Pull request by danielfleischer**
# Description

- New class is based on HuggingFace models that were optimized for Intel. 
- This is inference code only. 
- Dependencies required: optimum[exporters], optimum-Intel, neural-compressor.
- Added example notebook.

## Type of Change

- [X] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

- [ ] Added new unit/integration tests
- [X] Added new notebook (that tests end-to-end)
- [X] I stared at the code and made sure it makes sense

# Suggested Checklist:

- [X] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [ ] I have added Google Colab support for the newly added notebooks.
- [X] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] I ran `make format; make lint` to appease the lint gods

### CODE REVIEW

File: llama_index/embeddings/huggingface_intel.py [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+
+        if self.pooling == "cls":
+            embeddings = self._cls_pooling(model_output)
+        else:
+            embeddings = self._mean_pooling(
+                model_output, encoded_input["attention_mask"].to(self._device)
+            )
+
+        if self.normalize:
+            embeddings = torch.nn.functional.normalize(embeddings, p=2, dim=1)
```
Comment (hatianzhang) [inline llama_index/embeddings/huggingface_intel.py:126]: should we make `p` `dim` as part of class variable

File: pyproject.toml [hunk 2]
```diff
@@ -91,7 +93,9 @@ langchain = [
     "langchain",
 ]
 local_models = [
+    "neural-compressor",
     "optimum",
+    "optimum-intel",
```
Comment (hatianzhang) [inline pyproject.toml:98]: do we need to import those pkgs here. since you are lazy import them

Comment (danielfleischer) [reply pyproject.toml:98]: It's not either or, you can have everything "local" downloaded with `-E local_models` and if you don't, you get a helpful message.

File: llama_index/embeddings/huggingface_intel.py (hunk 1 above)
Comment (danielfleischer) [reply llama_index/embeddings/huggingface_intel.py:126]: I just took the same [code](https://github.com/danielfleischer/llama_index/blob/intel/llama_index/embeddings/huggingface.py#L172-L175) as in a general huggingface local model.

**Review by danielfleischer — COMMENTED** [review]
Made 2 comments; what do you think?

**Comment by logan-markewich** [comment]
@danielfleischer apologies, with v0.10.0 the project structure has changed slightly

I can take a stab at porting this PR in a bit, if you don't get to it first :)

**Comment by danielfleischer** [comment]
I'm having a look and modifying as needed.

**Comment by danielfleischer** [comment]
I forced push a re-implementation using the "integrations" format of v0.10, thanks!

**Review by hatianzhang — APPROVED** [review]
lgtm
